amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseRefresher.java

```diff
@@ -533,14 +533,22 @@ public boolean takeLease(@NonNull final Lease lease, @NonNull final String owner
         final AWSExceptionManager exceptionManager = createExceptionManager();
         exceptionManager.add(ConditionalCheckFailedException.class, t -> t);
 
-        Map<String, AttributeValueUpdate> updates = serializer.getDynamoLeaseCounterUpdate(lease);
-        updates.putAll(serializer.getDynamoTakeLeaseUpdate(lease, owner));
-
-        UpdateItemRequest request = UpdateItemRequest.builder().tableName(table).key(serializer.getDynamoHashKey(lease))
-                .expected(serializer.getDynamoLeaseCounterExpectation(lease)).attributeUpdates(updates).build();
+        GetItemRequest getItemRequest = GetItemRequest.builder()
+                .tableName(table)
+                .key(serializer.getDynamoHashKey(lease))
+                .build();
 
         try {
             try {
+                final GetItemResponse getItemResponse =
+                        FutureUtils.resolveOrCancelFuture(dynamoDBClient.getItem(getItemRequest), dynamoDbRequestTimeout);
+                final Lease currentLease = serializer.fromDynamoRecord(getItemResponse.item());
+
+                Map<String, AttributeValueUpdate> updates = serializer.getDynamoLeaseCounterUpdate(currentLease);
+                updates.putAll(serializer.getDynamoTakeLeaseUpdate(currentLease, owner));
```

Comment:
We need to have this update in DynamoDBLeaseTaker after `computeLeasesToTake`, to avoid querying lease table on each of the retry and to avoid other callers from updating every other time. 
Also we need to do this update only when the `updateAllLeases` takes more time than the lease renewal interval.